server/test/utils.ts

```diff
@@ -0,0 +1,94 @@
+import { rand, randEmail, randNumber, randUserName, randQuote, randBrand } from '@ngneat/falso';
+
+import { User } from 'server/models';
+
+// user interface
+interface UserObject {
+  email?: string;
+  discord_user_id?: string;
+  discord_name?: string;
+  bio?: string;
+  twitter_username?: string;
+  linkedin_url?: string;
+  github_username?: string;
+  website?: string;
+}
+
+// https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range/39495173#39495173
+// Tim's fault for telling me this is possible
+type Enumerate<N extends number, Acc extends number[] = []> = Acc['length'] extends N
+  ? Acc[number]
+  : Enumerate<N, [...Acc, Acc['length']]>;
+
+type IntRange<F extends number, T extends number> = Exclude<Enumerate<T>, Enumerate<F>>;
+
+// random
+// probability should only be 1 - 100
+function randomEmptyChance(probability: IntRange<0, 100>, preferredResult: string) {
+  // out of a 100
+  const randomNum = Math.random() * 100;
+  return randomNum < probability ? '' : preferredResult;
```

Comment:
What is overriding defaults?

Reply:
@Caleb-Cohen Overriding defaults means changing default behavior.

For instance, the `createUser` function accepts an object with each `User` field as a possible key. If a key doesn't exist in the object (or if no object is provided), it will generate a random values with the possibility of some fields being empty. That random value or empty would be the default behavior... passing in a specific value would be an override.

However, because `randomEmptyChance` isn't exported, a user can't pass in an override and maintain the possibility of it being empty. Probably not something you'd need often, but being able to replicate that empty chance from outside the utility functions would be useful. eg.
```js
// Extra long discriminator (unrealisitc example)
const username = `${randUser({ length: 10 })}#randNumber({ length: 6 });

createUser({
    // Retains the possibility of being empty
    discord_name: randomEmptyChance(20, username),
});
```

Reply:
This function is overriding defaults by giving a low chance (20% ) to return an empty string otherwise it will return the passed-in value, for all cases this value is another random function provided by falso for each related property. 

This function is being used on the userObject that is built by `createUser`

https://github.com/timmyichen/dev-directory/blob/d45bf4c12d28ce68af59adbab0ad0c3d152572df/server/test/utils.ts#L52-L61


An example of this with `twitter_username` 
(only the single property is shown for example purposes, but normally all the properties would be there) 

```ts
twitter_username: twitter_username ?? randomEmptyChance(20, randUserName())

// createUser({twitter_username: 'example'})
// output userObject = {twitter_username: 'example'}

// createUser()
// output userObject = {twitter_username: 'Joyce_Azulay85'}

// createUser()
// output if randomEmptyChance triggers userObject = {twitter_username: ''}
```